### PR TITLE
Add Buster containers for PHP 5.4 through 7.4

### PR DESCRIPTION
**Pull request by SammyK**
### Description

- Add Buster containers for PHP 5.4 through 7.4
- Update base Buster container with libmemcached, and default Catch2 & cmake installs
- Update PHP 8.0 Buster on new base & (+pcntl, -phpdbg)
- Fix a few places where `sha256sum` hashes weren't being checked properly
- Fix the "Permission denied" component tests fail on Buster

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: dockerfiles/ci/buster/Dockerfile [hunk 1]
```diff
@@ -81,22 +82,24 @@ RUN set -eux; \
     \
 # Allow nginx to be run as non-root for tests
     chown -R circleci:circleci /var/log/nginx/ /var/lib/nginx/; \
-# JSON parser
-    export JQ_URL="https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/jq-latest"; \
-    curl --silent --show-error --location --fail --retry 3 --output /usr/bin/jq $JQ_URL; \
-    chmod +x /usr/bin/jq; \
-    jq --version; \
-    \
+# Install CMake
```
Comment (morrisonlevi) [inline dockerfiles/ci/buster/Dockerfile:85]: Heads up, there's a new CMake release. You'll want something more like:
```
CMAKE_VERSION="3.20.1"
CMAKE_SHA256="b8c141bd7a6d335600ab0a8a35e75af79f95b837f736456b5532f4d717f20a09"
curl -OL https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz
(echo "${CMAKE_SHA256} cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz" | sha256sum -c -)
mkdir -p /opt/cmake/${CMAKE_VERSION}
tar --strip-components 1 -C /opt/cmake/${CMAKE_VERSION} -xf /tmp/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz;
```

File: dockerfiles/ci/buster/Dockerfile [hunk 2]
```diff
@@ -81,22 +82,24 @@ RUN set -eux; \
...
+    cd /tmp && curl -OL https://github.com/Kitware/CMake/releases/download/v3.19.6/cmake-3.19.6-Linux-x86_64.tar.gz; \
+    (echo "139580473b84f5c6cf27b1d1ac84e9aa6968aa13e4b1900394c50075b366fb15 cmake-3.19.6-Linux-x86_64.tar.gz" | sha256sum -c -); \
+    mkdir -p /opt/cmake/3.19.6 && cd /opt/cmake/3.19.6; \
+    tar -xf /tmp/cmake-3.19.6-Linux-x86_64.tar.gz --strip 1; \
+# Install Catch2
```
Comment (morrisonlevi) [inline dockerfiles/ci/buster/Dockerfile:90]: New catch2 version released 2 days ago as well.

**Review by morrisonlevi — COMMENTED** [review]
Heads up, there are newer versions of Catch2 and CMake.

File: dockerfiles/ci/buster/Dockerfile (hunk 1 above)
Comment (SammyK) [reply dockerfiles/ci/buster/Dockerfile:85]: Thanks. This is updated.

File: dockerfiles/ci/buster/Dockerfile (hunk 2 above)
Comment (SammyK) [reply dockerfiles/ci/buster/Dockerfile:92]: Nice _catch_. 😉 This is updated.

**Review by morrisonlevi — APPROVED** [review]
Approved. We should probably update the [readme](https://github.com/DataDog/dd-trace-php/blob/master/dockerfiles/ci/README.md) to use `docker-compose build --no-cache --pull` to update images like `composer:2` and `datadog/dd-trace-ci:alpine` as part of building to make sure we have the latest updates. Doesn't block this PR, though.

**Pull request merged**